### ISSUE TITLE
Construction macros don't work when nalgebra is aliased as na

### ISSUE THREAD

**Issue by nickeb96** [dimforge/nalgebra#1163]
If you alias `nalgebra` as `na` in Cargo.toml:

```
[dependencies]
...
na = { package = "nalgebra", version = "0.31", features = ["bytemuck"] }
...
```

The `point!` and `vector!` macros do not work.

Perhaps they internally use `nalgebra` instead of `$crate` as the path root?

**Comment by Andlon** [comment]
Indeed, that sounds about right. It's been a while since I wrote those macros - I'm not quite sure how to emit `$crate` from a proc macro correctly. If you are (or anyone else is) up for investigating this, it would be much appreciated!

**Comment by Ralith** [comment]
There's [`Span::mixed_site`](https://docs.rs/proc-macro2/1.0.46/proc_macro2/struct.Span.html#method.mixed_site), but that gets you the crate the *macro* was defined in, which isn't actually all that helpful here. Proc macros make this difficult.

**Comment by nickeb96** [comment]
It looks like there might be a way to reference `nalgebra` from `nalgebra-macros` regardless of whether it was renamed: <https://stackoverflow.com/a/57049687/4705242>

Another potential option is to have declarative macros act as wrappers around the procedural macros.  That way the declarative macros could use `$crate` and pass it in to the proc macro.

**Comment by Ralith** [comment]
A declarative wrapper seems like a viable solution.

> https://stackoverflow.com/a/57049687/4705242

This still has the same problem: it gets you the macro crate, not nalgebra.

**Comment by geo-ant** [comment]
I ran into the same problem a long-ish while ago and it seemed to me there was no standard solution to that problem. Someone (not me) has written [a crate](https://crates.io/crates/find-crate) that parses the `Cargo.toml` file at build time to find out what the own crate was imported as. I decided against using that though, because I don't know how robust this is. Maybe there's better alternatives out there by now, but the fact that someone can rename my crate when importing and break my macros has always made me a bit sad...
